sklearn/decomposition/pca.py

```diff
@@ -523,6 +534,13 @@ def _fit_truncated(self, X, n_components, svd_solver):
                                      flip_sign=True,
                                      random_state=random_state)
 
+        elif svd_solver == 'lobpcg':
+            # sign flipping is done inside
+            U, S, V = lobpcg_svd(X, n_components=n_components,
+                                 n_iter=self.iterated_power,
+                                 flip_sign=True,
+                                 random_state=random_state)
```

Comment:
This doesn't pass tol

Reply:
The new lobpcg code internally substitutes tol = .0 or None with a "reasonable" default, so there is no need to pass tol.